src/Core/Form/IdentifiableObject/CommandBuilder/Product/SEOCommandsBuilder.php

```diff
@@ -44,23 +47,55 @@ public function buildCommands(ProductId $productId, array $formData): array
 
         $seoData = $formData['seo'] ?? [];
         $redirectionData = $formData['seo']['redirect_option'] ?? [];
+
+        $commands = [];
         $command = new UpdateProductSeoCommand($productId->getValue());
+        $seoUpdate = false;
 
         if (isset($seoData['meta_title'])) {
             $command->setLocalizedMetaTitles($seoData['meta_title']);
+            $seoUpdate = true;
         }
         if (isset($seoData['meta_description'])) {
             $command->setLocalizedMetaDescriptions($seoData['meta_description']);
+            $seoUpdate = true;
         }
         if (isset($seoData['link_rewrite'])) {
             $command->setLocalizedLinkRewrites($seoData['link_rewrite']);
+            $seoUpdate = true;
         }
 
         if (isset($redirectionData['type'])) {
             $targetId = (int) ($redirectionData['target']['id'] ?? 0);
             $command->setRedirectOption($redirectionData['type'], $targetId);
+            $seoUpdate = true;
+        }
+
+        // Only return the command if some updates have been detected
+        if ($seoUpdate) {
```

Comment:
Only a suggestion about naming and strict comparison, nothing mandatory :)
```suggestion
        if ($isSeoUpdated === true) {
```

Reply:
This new behaviour might be complex to implement with the new multi-shop command builder. Any idea ?

Reply:
Actually, it won't, this is already handled internally by the command builder The commands are systematically instantiated and passed to the command builder

However, if it doesn't detect any fields in the data array the command is not returned and is "filtered" so it's the same principle as here actually, but easier 😉

Reply:
And I don't think the strict test is needed here, besides as you mentioned Your PR will replace this boolean via the CommandBuilder

Do you think I should refacto my PR to match yours and thus limit the conflicts later? Regardless of which one is merged first?